design/repo_maintenance_job_config.md

```diff
@@ -0,0 +1,144 @@
+# Repository maintenance job configuration design
+
+## Abstract
+Add this design to make the repository maintenance job can read configuration from a dedicate ConfigMap and make the Job's `PodSpec.Affinity` configurable. 
+
+## Background
+Repository maintenance is split from the Velero server to a k8s Job in v1.14 by [repository maintenance job](Implemented/repository-maintenance.md).
+The repository maintenance Job configuration was read from the Velero server CLI parameter.
+The Job inherits the most of the Job's PodSpec configuration from Velero server's Deployment's PodSpec.
+That inheritance also includes the Affinity setting.
+
+[node-agent affinity configuration](Implemented/node-agent-affinity.md) introduces affinity configuration for the node-agent DaemonSet pods.
+This design reuses the data structure introduced by that design to make the repository maintenance job can choose which node running on.
+
+## Goals
+- Let user can choose repository maintenance Job running on which nodes.
+- Unify the repository maintenance Job configuration at one place.
+
+## Non Goals
+- There was an [issue](https://github.com/vmware-tanzu/velero/issues/7911) to require the whole Job's PodSpec should be configurable. That's not in the scope of this design.
+- The introduced configuration is universal for all repositories. Configuration dedicated for each repository is not in this design's scope.
+
+## Alternatives Considered
+An other option is making each BackupRepository having its own repository maintenance job configuration.
+Each BackupRepository has its own storage capacity, so they may require different resource to maintain. 
+This can be done by adding a field to the BackupRepository CRD to specify the referenced ConfigMap.
```

Comment:
> This can be done by adding a field to the BackupRepository CRD to specify the referenced ConfigMap.

It is better we can make a clear differentiation of what go into repository maintenance job configuration and what go into backupRepository CR/configuration.
I think the resource limit should go into repository maintenance job configuration, so backupRepository CR/configuration should not take it in anyway.
If we think configuring different resource limits for different repo is a requirement, we should not exclude the future plan that making the repository maintenance job configuration backup repo specific.